File: python-automation/src/automation/pages/login_page.py
from playwright.sync_api import Page
from src.automation.data.test_data import Users, Urls, ErrorMessages


class LoginPage:
    def __init__(self, page: Page):
        self.page = page

    @property
    def username_input(self):
        return self.page.locator("[data-test='username']")

    @property
    def password_input(self):
        return self.page.locator("[data-test='password']")

    @property
    def login_button(self):
        return self.page.locator("[data-test='login-button']")

    @property
    def error_message(self):
        return self.page.locator("[data-test='error']")

    def go_to(self):
        self.page.goto(Urls.BASE)

    def login(self, username: str, password: str):
        self.username_input.fill(username)
        self.password_input.fill(password)
        self.login_button.click()

    def login_and_expect_error(self, username: str, password: str, expected_text: str):
        self.login(username, password)
        assert self.error_message.is_visible(), "Error message should be visible"
        assert expected_text in self.error_message.text_content(), \
            f"Error message should contain: {expected_text}"

    def assert_logged_in(self):
        assert self.page.url == Urls.INVENTORY, \
            f"User should be redirected to {Urls.INVENTORY}, but got {self.page.url}"

File: java-automation/src/main/java/com/automation/pages/LoginPage.java
package com.automation.pages;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.Locator;
import com.automation.data.TestData;

import static org.junit.jupiter.api.Assertions.*;

public class LoginPage {
    private final Page page;

    public LoginPage(Page page) {
        this.page = page;
    }

    private Locator getUsernameInput() {
        return page.locator("[data-test='username']");
    }

    private Locator getPasswordInput() {
        return page.locator("[data-test='password']");
    }

    private Locator getLoginButton() {
        return page.locator("[data-test='login-button']");
    }

    private Locator getErrorMessage() {
        return page.locator("[data-test='error']");
    }

    public void goTo() {
        page.navigate(TestData.Urls.BASE);
    }

    public void login(String username, String password) {
        getUsernameInput().fill(username);
        getPasswordInput().fill(password);
        getLoginButton().click();
    }

    public void loginAndExpectError(String username, String password, String expectedText) {
        login(username, password);
        assertTrue(getErrorMessage().isVisible(), "Error message should be visible");
        assertTrue(getErrorMessage().textContent().contains(expectedText),
            "Error message should contain: " + expectedText);
    }

    public void assertLoggedIn() {
        assertEquals(TestData.Urls.INVENTORY, page.url(), "User should be redirected to inventory page");
    }

    /** Assert the X icon exists on the error banner (visual cue for recruiters) */
    public void assertErrorIconVisible() {
        Locator icon = getErrorMessage().locator("svg[data-icon='xmark']");
        assertTrue(icon.isVisible(), "Error icon should be visible on the error banner");
    }
}
